### PR TITLE
Switch use from `JacksonFactory` to `GsonFactory`.

### PR DESCRIPTION
**Pull request by Capstan**
This is mirroring a global change we're making in the monorepo to reduce concerns from security.

### CODE REVIEW

**Comment by Capstan** [comment]
Gradle is strangely reporting a seemingly-unrelated Guava error when I build with these changes:

```
> Task :portability-transfer:compileJava FAILED
/usr/local/google/home/nherring/github/data-transfer-project/portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerModule.java:18: error: cannot find symbol
import static com.google.common.collect.MoreCollectors.onlyElement;
                                       ^
  symbol:   class MoreCollectors
  location: package com.google.common.collect
/usr/local/google/home/nherring/github/data-transfer-project/portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerModule.java:18: error: static import only from classes and interfaces
import static com.google.common.collect.MoreCollectors.onlyElement;
^
2 errors
```

I cannot tell if my depends are perturbing something in the Guava version resolution machine.

**Review by jzacsh — COMMENTED** [review]
Left a comment but otherwise LGTM. I may hold this off just to do some internal testing.

**Comment by Capstan** [comment]
`./gradlew test` from the root is what I was running.

**Comment by Capstan** [comment]
And the `demo-server` `build.gradle` blanket overrides the guava version for all artifacts, which doesn't work for the versioning scheme for `listenablefuture` or `failureaccess`.

File: distributions/demo-server/build.gradle
Comment (Capstan) [inline distributions/demo-server/build.gradle:202]: I wonder if having this `resolutionStrategy` in `portability-transfer/build.gradle` would also have worked around the original problem?

This whole thing seems quite fragile.

File: portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerMain.java
Comment (seehamrun) [inline portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerMain.java:80]: I have a slight concern with updating this in the extension context because it gets pulled in arbitrarily in other classes.. ex: it gets piped through to the GooglePhotosInterface through this line: https://github.com/google/data-transfer-project/blob/889f1fa92fc54046b42a9448b53d9de0df84912c/extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/GoogleTransferExtension.java#L76,L109 

I'm not sure how this would behave if another service is using Jackson somewhere else in their code but pulling in this Gson factory instead. 

I would guess there shouldnt be too many instances of this but i have not done the analysis

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java
Comment (seehamrun) [inline extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java:137]: I wonder if we can just use `this.jsonFactory` instead of creating a new instance here and below on L171?

Comment (Capstan) [reply extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java:137]: That does seem like a better approach overall.

Comment (Capstan) [reply extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java]: Done.

File: distributions/demo-server/build.gradle
Comment (seehamrun) [reply distributions/demo-server/build.gradle:202]: Hiya! sorry for the delay on this review, I was discussing this with zacsh@ yesterday and there are a few things that arent clear with this... 

1) where/how did you determine that you also needed `failureaccess` and `listenablefuture` here? Its not obvious from the logs 
2) If we delete this block entirely, would everything just work? 
3) Im looking through to see if there are other places where we explicitly make a new JacksonFactory, in theory there shouldn't be since each class should use the one that's registered in the extensioncontext in WorkerMain.java, but we've seen that the GooglePhotosInterface broke that :) so we should double check the project for other instances of this.

Comment (Capstan) [reply distributions/demo-server/build.gradle:202]: If you remove them, you end up with failures to find those targets at the overridden version when you use the latest guava (which includes them and includes versioned references to them, and those versions get edited, IIRC).

File: portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerMain.java
Comment (Capstan) [reply portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerMain.java:80]: I updated GooglePhotosInterface to use the provided `JsonFactory` so it will at least be consistent with whatever is done here.

File: distributions/demo-server/build.gradle
Comment (Capstan) [reply distributions/demo-server/build.gradle:202]: For 2; I don't know. I have definitely seen weird compilation failures related to be should-be-findable-but-aren't methods in `MoreCollections`. I don't understand why version resolution doesn't force a later version of Guava that has them naturally, but I haven't fully internalized gradle/Java's semantics here. I can try it later, but I'm OOO for the next 4d.

For 3, I ran a bulk editor in the monorepo that looked at `JacksonFactory` use, but it appears not to have found all the references in here, so I'll need to do a second pass manually.

Comment (Capstan) [reply distributions/demo-server/build.gradle:202]: Had more time (and battery) than I thought, and I tried out removing the entire `resolutionStrategy` block, and that seems to have worked, so I'm going to roll that into the guava update commit.

**Comment by Capstan** [comment]
Ugh. `git rebase -i` is not my friend. I redid the edits three different times and still managed to drop them on the floor. :(

File: distributions/demo-server/build.gradle
Comment (jzacsh) [reply distributions/demo-server/build.gradle]: > For 3, I ran a bulk editor in the monorepo that looked at JacksonFactory use, but it appears not to have found all the references in here, so I'll need to do a second pass manually.

For context on this: we don't import the _entirety_ of this codebase into the monorepo (in case you have any other cases like ours in your lsc seeing only subsets of stuff).

**Review by seehamrun — APPROVED** [review]
this LGTM! Thanks for doing this cleanup

**Comment by Capstan** [comment]
Is this waiting on an affirmative review by @jzacsh and/or @wmorland, or was that advisory only? In the latter case, how long until it's worth merging and dealing with potential aftermath of a late review (if any) later?

**Comment by Capstan** [comment]
Still waiting on a review.

**Review by wmorland — APPROVED** [review]
Reviewed and looks good to me. From our side the biggest change may be the guava upgrade which we might require to propogate.

**Pull request merged**
